src/routing/core/messaging/handling/bad_msgs.rs

```diff
@@ -82,11 +82,11 @@ impl Core {
                         section,
                         network: network.clone(),
                     },
-                    None,
+                    self.section.authority_provider().section_key,
                 )?
             }
             _ => bounced_msg
-                .extend_proof_chain(&dst_key, self.section.chain())
+                .updated_with_latest_key(self.section.authority_provider().section_key)
```

Comment:
for `AggregationAtSrc` message, updating to a new section_key means they have to carry out a new round of vote using the new section_key_share .

Reply:
We only send it back with just the latest keys that we know since the original message could have been sent with a later key and is now being bounced back,